classes/Models/Occurrence.php

```diff
@@ -44,12 +44,14 @@ public function GetMetaValue($name, $default = array())
      */
     public function SetMetaValue($name, $value)
     {
-        //get meta adapter
-        $model = new WSAL_Models_Meta();
-        $model->occurrence_id = $this->getId();
-        $model->name = $name;
-        $model->value = maybe_serialize($value);
-        $model->SaveMeta();
+        if (isset($value)) {
```

Comment:
@carmelopreeostudios i think this will always be true since it's a parameter - you should use !empty if you do not want to execute the code if the value is an empty string or null